be/src/Project/Services/Inventory/Inventory.Api/v1/Controllers/InventoryController.cs

```diff
@@ -194,6 +194,46 @@ public async Task<IActionResult> Get(string ids, string searchTerm, int? pageInd
             return this.StatusCode((int)HttpStatusCode.UnprocessableEntity);
         }
 
+        [HttpGet, MapToApiVersion("1.0")]
+        [Route("availableproductssuggestions")]
+        [ProducesResponseType((int)HttpStatusCode.OK, Type = typeof(IEnumerable<string>))]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> GetAvailableProductsInventorySuggestions(
+            string searchTerm, int suggestionsCount)
+        {
+            var sellerClaim = User.Claims.FirstOrDefault(x => x.Type == AccountConstants.Claims.OrganisationIdClaim);
+
+            var serviceModel = new GetInventorySuggestionsServiceModel
+            {
+                SearchTerm = searchTerm,
+                SuggestionsCount = suggestionsCount
+            };
+
+            var validatior = new GetInventorySuggestionsModelValidator();
+            var validationResult = await validatior.ValidateAsync(serviceModel);
+
+            if (validationResult.IsValid)
+            {
+                var inventorySuggestions = _inventoriesService.GetInventorySuggestions(serviceModel);
```

Comment:
**suggestion:** I would simplify this controller method. Look at others controllers